### DISCUSSION TITLE
Is there a configuration option for fullClone?

### DISCUSSION THREAD

**Discussion by chopeen** [renovatebot/renovate#30956] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab + Renovate image `38.46-full`

### Please tell us more about your question or problem

What the configuration option that corresponds with `config.fullClone` below?

https://github.com/renovatebot/renovate/blob/65b42128b67ffe8bdec96d89fbcd963fa361304f/lib/util/git/index.ts#L417-L422

I checked [configuration-options](https://docs.renovatebot.com/configuration-options) and [self-hosted-configuration](https://docs.renovatebot.com/self-hosted-configuration), I did not find any.

---

I have a certain repository that gets cloned and then the scanning process shows a series of _has no content_ debug messages for all the files:

```
2024-08-20T10:02:44.988330Z 01O DEBUG: vue_frontend/src/store/mutations.js has no content
2024-08-20T10:02:44.988453Z 01O DEBUG: vue_frontend/src/store/otss/actions.js has no content
2024-08-20T10:02:44.988587Z 01O DEBUG: vue_frontend/src/store/otss/getters.js has no content
2024-08-20T10:02:44.988710Z 01O DEBUG: vue_frontend/src/store/otss/index.js has no content
```

I would like to test if changing the cloning mode could fix that.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
There is no such exposed configuration option today

**Reply by darlanalves** [reply]
As an alternative, you can do the git clone outside of Renovate, then point the tool to it.

If you look at the same file listed above, before the code enters the clone path, it will first check for the presence of `.git/HEAD` inside the current working repository.

If Renovate is working on an existing full clone, then only a `git fetch/pull`, followed by a few resets, will happen instead.

**Reply by darlanalves** [reply]
If you run with Docker it's easy. Add a volume with your cloned repo:

`docker run ... -v $PWD:/tmp/renovate/repos/gitlab/org/repo`

**Reply by viceice** [reply]
you should analyze why the content is missing instead of doing hacky workarounds.

**Reply by chopeen** [reply, marked as answer]
After some further digging, I can tell this message had **nothing to do with cloning the code**.

It is logged at [manager-files.ts#L71](https://github.com/renovatebot/renovate/blob/85c1163e2e7f443bc6a42fb565e3e636daf88649/lib/workers/repository/extract/manager-files.ts#L71) and there is a message for every single file in the repository, because I have the custom [aws-machine-image](https://docs.renovatebot.com/modules/datasource/aws-machine-image) manager configured with `fileMatch: [".*"]`. This custom manager loads every file and failing to find any AMIs logs the message `has no content`.
